haystack/core/pipeline/pipeline.py

```diff
@@ -390,6 +395,24 @@ def run(  # noqa: PLR0915, PLR0912, C901, pylint: disable=too-many-branches
                 except PipelineRuntimeError as error:
                     # Attach partial pipeline outputs to the error before re-raising
                     error.pipeline_outputs = pipeline_outputs
+                    # Create a snapshot of the last good state of the pipeline before the error occurred.
+                    last_good_state_snapshot = self.create_last_good_state_snapshot(
+                        component_inputs,
+                        component_name,
+                        component_visits,
+                        data,
+                        include_outputs_from,
+                        inputs,
+                        ordered_component_names,
+                        pipeline_outputs,
+                    )
+                    f_name = f"last_good_state_{datetime.now().strftime('%Y_%m_%d_%H_%M_%S')}.json"
+                    _save_pipeline_snapshot(pipeline_snapshot=last_good_state_snapshot, f_name=f_name)
+                    logger.info(
+                        "A snapshot of the last good state of the pipeline has been saved to '{f_name}'. "
+                        "You can inspect this snapshot to fix the error and resume the pipeline from this point.",
+                        f_name=f_name,
+                    )
```

Comment:
Let's slightly reword this
```suggestion
                    logger.info(
                        "Saved a snapshot of the pipeline's last valid state to '{f_name}'. "
                        "Review this snapshot to debug the error and resume the pipeline from here.",
                        f_name=f_name,
                    )
```